### PR TITLE
fix: babylon private key conversion

### PR DESCRIPTION
**Pull request by AbstractFruitFactory**

### CODE REVIEW

**Review by CyonAlexRDX — APPROVED** [review]
LGTM

File: packages/crypto/package.json [hunk 1]
```diff
@@ -37,6 +37,7 @@
   "dependencies": {
     "@radixdlt/data-formats": "1.0.30",
     "@radixdlt/primitives": "^3.0.9",
+    "@radixdlt/radix-engine-toolkit": "^1.0.2",
```
Comment (dawidsowardx) [inline packages/crypto/package.json:40]: shouldn't it be `devDependency` if it's used only in tests?

File: packages/crypto/src/utils.ts [hunk 2]
```diff
@@ -53,3 +53,10 @@ export const validateLength = (
 		  )
 		: ok(buffer)
 }
+
+export const buf2hex = function (buffer: ArrayBuffer) {
```
Comment (LukasGasior1) [inline packages/crypto/src/utils.ts:57]: We should prolly rename it to make it clear that it also reverses the array (I believe bytes being stored in reverse is specific to uint256, not a JS thing?). Or reverse before calling it

Comment (dhedey) [reply packages/crypto/src/utils.ts:57]: Right, perhaps the method would be more clearly abstracted as `toPrivateKeyHex(scalar: UInt256): string`

File: packages/crypto/test/messageEncryption.test.ts [hunk 3]
```diff
@@ -213,4 +222,39 @@ describe('message encryption', () => {
 		//	)
 		//})
 	})
+
+	it('should convert to a valid babylon private key', async () => {
```
Comment (dhedey) [inline packages/crypto/test/messageEncryption.test.ts:226]: This test doesn't seem relevant to the issue at hand.

Even the old toString would have satisfied this test.

A better test is that `babylonKey.toString() === olympiaKey.toString()`

File: packages/crypto/test/messageEncryption.test.ts [hunk 4]
```diff
@@ -213,4 +222,39 @@ describe('message encryption', () => {
...
+		const hdMasterSeed = c.HDMasterSeed.fromMnemonic({ mnemonic })
+
+		const key = a.SigningKey.fromHDPathWithHDMasterSeed({
+			hdPath: c.HDPathRadix.fromString(HD_PATH)._unsafeUnwrap(),
+			hdMasterSeed
+		})
+
+		const hdKey = hdMasterSeed.masterNode().derive(c.HDPathRadix.fromString(HD_PATH)._unsafeUnwrap())
+
+		const privateKey = new PrivKey.Secp256k1(hdKey.privateKey.toString())
```
Comment (dhedey) [inline packages/crypto/test/messageEncryption.test.ts:240]: Perhaps we should export it as `BabylonPrivateKey`?

File: packages/crypto/test/messageEncryption.test.ts (hunk 3 above)
Comment (AbstractFruitFactory) [reply packages/crypto/test/messageEncryption.test.ts:226]: this seemed to me to be the root issue? the test also did fail with the old method. But sure, I can update it.
